packages/tailwindcss/src/utilities.test.ts

```diff
@@ -22056,6 +22606,89 @@ test('backdrop-filter', async () => {
       'backdrop-sepia-[var(--value)]/foo',
     ]),
   ).toEqual('')
+
+  expect(
+    await compileCss(
+      css`
+        @theme {
+          --backdrop-blur-none: 2px;
+        }
+        @tailwind utilities;
+      `,
+      ['backdrop-blur-none'],
+    ),
+  ).toMatchInlineSnapshot(`
+    "@layer properties {
+      @supports (((-webkit-hyphens: none)) and (not (margin-trim: inline))) or ((-moz-orient: inline) and (not (color: rgb(from red r g b)))) {
+        *, :before, :after, ::backdrop {
+          --tw-backdrop-blur: initial;
+          --tw-backdrop-brightness: initial;
+          --tw-backdrop-contrast: initial;
+          --tw-backdrop-grayscale: initial;
+          --tw-backdrop-hue-rotate: initial;
+          --tw-backdrop-invert: initial;
+          --tw-backdrop-opacity: initial;
+          --tw-backdrop-saturate: initial;
+          --tw-backdrop-sepia: initial;
+        }
+      }
+    }
+
+    :root, :host {
+      --backdrop-blur-none: 2px;
+    }
+
+    .backdrop-blur-none {
+      --tw-backdrop-blur: blur(var(--backdrop-blur-none));
+      -webkit-backdrop-filter: var(--tw-backdrop-blur, ) var(--tw-backdrop-brightness, ) var(--tw-backdrop-contrast, ) var(--tw-backdrop-grayscale, ) var(--tw-backdrop-hue-rotate, ) var(--tw-backdrop-invert, ) var(--tw-backdrop-opacity, ) var(--tw-backdrop-saturate, ) var(--tw-backdrop-sepia, );
+      backdrop-filter: var(--tw-backdrop-blur, ) var(--tw-backdrop-brightness, ) var(--tw-backdrop-contrast, ) var(--tw-backdrop-grayscale, ) var(--tw-backdrop-hue-rotate, ) var(--tw-backdrop-invert, ) var(--tw-backdrop-opacity, ) var(--tw-backdrop-saturate, ) var(--tw-backdrop-sepia, );
+    }
```

Comment:
Not related to this PR, but this output is crazy and made me think about something... I wonder if we can be a bit smarter here and optimize the output based on what you used.

So if you _only_ use `blur` that we get rid of all the other variables used here.

If you have a utility that sets some backdrop opacity, or if we detect that `--tw-backdrop-saturate` is used, that we start including it in the output.

Food for thought 🤔

Reply:
Hmm, interesting, but unfortunately would be a breaking change if people used runtime to define these variables right? 🤔

Reply:
Yeah fair I guess 👍

Reply:
Yeah it does feel like a bit of a stretch, especially in regard to the changes in this PR.

Reply:
Yep but I like to think about it in a different way. Similar to how we define CSS variables and remove them when they are not used, or similar to how we only emit CSS for used utilities today.

The CSS variables are also only emitted based on the presence of _other_ utilities (except for `@theme static`).

If you think about it like this, then it could be part of the optimization step where we remove certain `@property` at-rules or used `var(--tw-*)` in declarations. Might not be as easy to generalize this but it wouldn't touch the cached AST.

Reply:
> Yeah it does feel like a bit of a stretch, especially in regard to the changes in this PR.

Hence the `Not related to this PR` part, just noticed it while looking over the diff 😅